### PR TITLE
Fix ocamlfind.1.9.6 upgrade

### PR DESCRIPTION
**Pull request by dra27**
Upgrading from ocamlfind 1.9.5 to 1.9.6 is failing on OCaml 5 switches. If you run:

```
opam switch create 5.0 ocaml-base-compiler.5.0.0
opam install ocamlfind.1.9.5
opam install ocamlfind.1.9.6
```

and then run `ocamlfind list` then none of the compiler's packages are included. This is fixed in https://github.com/ocaml/ocamlfind/pull/61, but as the fix only affects the `configure` file, it seems reasonable to patch it immediately in opam-repository.

Note that there is a workaround - `opam reinstall ocamlfind.1.9.6` permanently fixes the problem.

### CODE REVIEW

File: packages/ocamlfind/ocamlfind.1.9.6-1/opam [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  [make "all"]
+  [make "opt"] {ocaml:native}
+]
+install: [
+  [make "install"]
+  ["install" "-m" "0755" "ocaml-stub" "%{bin}%/ocaml"] {ocaml:preinstalled}
+]
+dev-repo: "git+https://github.com/ocaml/ocamlfind.git"
+url {
+  src: "http://download.camlcity.org/download/findlib-1.9.6.tar.gz"
```
Comment (mseri) [inline packages/ocamlfind/ocamlfind.1.9.6-1/opam:41]: The patch is missing

File: packages/ocamlfind/ocamlfind.1.9.6/opam [hunk 2]
```diff
@@ -13,7 +13,7 @@ authors: "Gerd Stolpmann <gerd@gerd-stolpmann.de>"
 homepage: "http://projects.camlcity.org/projects/findlib.html"
 bug-reports: "https://github.com/ocaml/ocamlfind/issues"
 depends: [
-  "ocaml" {>= "4.00.0"}
+  "ocaml" {>= "3.08.0"}
```
Comment (mseri) [inline packages/ocamlfind/ocamlfind.1.9.6/opam:16]:
```suggestion
  "ocaml" {>= "3.08.0" & < "5.0"}
```
Should we make this unavailable on ocaml 5?

**Comment by dra27** [comment]
All hopefully done. I also added a disabling commit for Windows (I'm separately looking into what's wrong there, but that will almost certainly need a new release)

File: packages/ocamlfind/ocamlfind.1.9.6-1/opam [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+authors: "Gerd Stolpmann <gerd@gerd-stolpmann.de>"
+homepage: "http://projects.camlcity.org/projects/findlib.html"
+bug-reports: "https://github.com/ocaml/ocamlfind/issues"
+depends: [
+  "ocaml" {>= "3.08.0"}
+]
+depopts: ["graphics"]
+build: [
+  # See https://github.com/ocaml/ocamlfind/pull/61
+  ["sed" "-ib" "s|stdlib/META|stdlib.cmi|" "configure"]
```
[the remark this answers is not in the document]
Comment (dra27) [reply packages/ocamlfind/ocamlfind.1.9.6-1/opam]: Changed to a patch - `sed` is used in a few (older) compiler packages, but I agree that it's bad to have a _latest_ version doing this!

**Comment by kit-ty-kate** [comment]
Thanks a lot

**Comment by ejgallego** [comment]
Hi @dra27 , I'm having some ocamfind windows issues on Opam 2.2, with 1.9.5 ; I see 1.9.6 is still disabled.

Do you have a pointer to the issues you found, to see if there would be a duplicate?

**Pull request merged**